### DISCUSSION TITLE
How to use Exogenous Variables with AutoNHITS or AutoTFT?

### DISCUSSION THREAD

**Discussion by blackhole-canada** [Nixtla/neuralforecast#1160] [Q&A] [answered]
Hello, I am trying to use some future, historic and static exogenous variables with NHITS and TFT models while doing hyperparameter optimization. However, I get an error about unexpected arguements while using exogenous variables with automodels while it works fine for normal NHITS or TFT models. Is there a way to use the exogenous variables for the auto models? Thank you in advance.

**Comment by elephaint** [comment, marked as answer]
Hi, you can do:

```
config = dict(input_size=12, hist_exog_list=[], stat_exog_list=[], futr_exog_list=[])
model = AutoNHITS(h=12, config=config)
```

**Reply by supriamir** [reply]
Hi @elephaint, i want to ask regarding to fine tuning and exogenous variables,



nhits_config = {
       "learning_rate": tune.choice([1e-3]),                                    
       "max_steps": tune.choice([1000]),                                       
       "input_size": tune.choice([2 * horizon]),                               
       "batch_size": tune.choice([7]),                                          
       "windows_batch_size": tune.choice([256]),                                
       "n_pool_kernel_size": tune.choice([[2, 2, 2], [16, 8, 1]]),               
       "n_freq_downsample": tune.choice([[168, 24, 1], [24, 12, 1], [1, 1, 1]]), 
       "activation": tune.choice(['ReLU']),                                      
       "n_blocks":  tune.choice([[1, 1, 1]]),                                  
       "mlp_units":  tune.choice([[[512, 512], [512, 512], [512, 512]]]),       
       "interpolation_mode": tune.choice(['linear']),                           
       "val_check_steps": tune.choice([100]),                                  
       "random_seed": tune.randint(1, 10),
    }

models = [
          AutoTFT(
              h=horizon,                 
              loss=MAE(),
              config=config,
              num_samples=10,
              search_alg=HyperOptSearch(),
               backend='ray',
            ),        
         ]

How can i use fine tuning and exogenous variable at the same time? Because in documentation, parameter config is used for hyperparameter optimization.

**Comment by blackhole-canada** [comment]
Thank you so much, it worked fine :)
